src/Loggers/OwnJSONPatternFormatter.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+
+#include <Poco/PatternFormatter.h>
+#include "ExtendedLogChannel.h"
+
+
+/** Format log messages own way in JSON.
+  * We can't obtain some details using Poco::PatternFormatter.
+  *
+  * Firstly, the thread number here is peaked not from Poco::Thread
+  * threads only, but from all threads with number assigned (see ThreadNumber.h)
+  *
+  * Secondly, the local date and time are correctly displayed.
+  * Poco::PatternFormatter does not work well with local time,
+  * when timestamps are close to DST timeshift moments.
+  * - see Poco sources and http://thread.gmane.org/gmane.comp.time.tz/8883
+  *
+  * Also it's made a bit more efficient (unimportant).
+  */
+
+class Loggers;
+
+class OwnJSONPatternFormatter : public Poco::PatternFormatter
```

Comment:
Does it make sense to inherit this class from OwnPatternFormatter and keep the interface the same? In this case, we would not need to change anything in OwnFormattingChannel and the way of choosing the correct formatter becomes a lot easier, for example:

```
...
Poco::AutoPtr<OwnPatternFormatter> pf;
if (config.has("logger.json"))
    pf = new OwnPatternFormatter(color_enabled);
else:
    pf = new OwnJSONPatternFormatter();

Poco::AutoPtr<DB::OwnFormattingChannel> log = new DB::OwnFormattingChannel(pf, channel);
...
```

And it becomes crucial for code simplicity if it is planned to add more than JSON format.

Reply:
Great suggestion! Made changes and pushed the code.